addons/maintainence-agent/maintenance_mode_controller.go

```diff
@@ -83,37 +72,63 @@ func (r *MaintenanceModeReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	}
 
 	if !mmode.GetDeletionTimestamp().IsZero() || len(mmode.Spec.Modes) == 0 {
-		result, err := r.startMaintenanceActions(ctx, mmode, actionableCephClusters, false)
+		if !utils.ContainsString(mmode.GetFinalizers(), MaintenanceModeFinalizer) {
```

Comment:
If there was no finalizer, then ignore the reconcile request. if deleted, or the MModes changed, and we never acted on the prior instance (which would have added the finalizer), then no point processing it.